api/files/serializers.py

```diff
@@ -285,20 +286,10 @@ def get_size(self, obj):
         return None
 
     def get_date_modified(self, obj):
-        mod_dt = None
-        if obj.provider == 'osfstorage' and obj.versions.exists():
-            # Each time an osfstorage file is added or uploaded, a new version object is created with its
-            # date_created equal to the time of the update.  The external_modified is the modified date
-            # from the backend the file is stored on.  This field refers to the modified date on osfstorage,
-            # so prefer to use the created of the latest version.
-            mod_dt = obj.versions.first().created
-        elif obj.provider != 'osfstorage' and obj.history:
-            mod_dt = obj.history[-1].get('modified', None)
-
-        if self.context['request'].version >= '2.2' and obj.is_file and mod_dt:
-            return datetime.strftime(mod_dt, '%Y-%m-%dT%H:%M:%S.%fZ')
-
-        return mod_dt and mod_dt.replace(tzinfo=pytz.utc)
+        date_modified = obj.versions.aggregate(Max('created'))['created__max']
+        if date_modified and self.context['request'].version >= '2.2' and obj.is_file:
```

Comment:
bad version comparison here. Also, switching to just the annotated result is going to break this versioning, which makes me think we need to provide a new value :(